presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/ExtendedHiveMetastore.java

```diff
@@ -101,7 +101,7 @@ default void dropTableFromMetastore(MetastoreContext metastoreContext, String da
 
     Optional<List<String>> getPartitionNames(MetastoreContext metastoreContext, String databaseName, String tableName);
```

Comment:
We should change this as well to return `Optional<List<PartitionNameWithVersion>>`